docs/guides/migrating-from-camunda-platform-7/adjusting-source-code.md

```diff
@@ -0,0 +1,82 @@
+- Remove existing Camunda Platform 7 settings.
+
+3. Replace `@EnableProcessApplication` with `@EnableZeebeClient` in your main Spring Boot application class.
+4. Add `@ZeebeDeployment(resources = "classpath*:**/*.bpmn")` to automatically deploy all BPMN models.
+
+Finally, adjust your source code and process model as described in the sections below.
+
+### Client API
+
+All Camunda Platform 8 APIs (e.g. to start process instances, subscribe to tasks, or complete them) have been completely redesigned are not compatible with Camunda Platform 7. While conceptually similar, the APIs use different method names, data structures, and protocols.
+
+If this affects large parts of your code base, you could write a small abstraction layer implementing the Camunda Platform 7 API delegating to Camunda Platform 8, probably marking unavailable methods as deprecated. We welcome community extensions that facilitate this but have not yet started own efforts.
+
+### Service tasks with attached Java code (Java Delegates, Expressions)
+
+In Camunda Platform 7, there are three ways to attach Java code to service tasks in the BPMN model using different attributes in the BPMN XML:
+
+- Specify a class that implements a JavaDelegate or ActivityBehavior: `camunda:class`.
+- Evaluate an expression that resolves to a delegation object: `camunda:delegateExpression`.
+- Invoke a method or value expression: `camunda:expression`.
+
+Camunda Platform 8 cannot directly execute custom Java code. Instead, there must be a [job worker](/components/concepts/job-workers.md) executing code.
+
+The [Camunda Platform 7 Adapter](https://github.com/camunda-community-hub/camunda-7-to-8-migration/tree/main/camunda-7-adapter) implements such a job worker using [Spring Zeebe](https://github.com/camunda-community-hub/spring-zeebe). It subscribes to the task type `camunda-7-adapter`. [Task headers](/components/modeler/bpmn/service-tasks/service-tasks.md#task-headers) are used to configure a delegation class or expression for this worker.
+
+![Service task in Camunda Platform 7 and Camunda Platform 8](../img/migration-service-task.png)
+
+You can use this worker directly, but more often it might serve as a starting point or simply be used for inspiration.
+
+The [Camunda Platform 7 to Camunda Platform 8 Converter](https://github.com/camunda-community-hub/camunda-7-to-8-migration/tree/main/backend-diagram-converter) will adjust the service tasks in your BPMN model automatically for this adapter.
+
+The topic `camunda-7-adapter` is set and the following attributes/elements are migrated and put into a task header:
+
+- `camunda:class`
+- `camunda:delegateExpression`
+- `camunda:expression` and `camunda:resultVariable`
+
+### Service tasks as external tasks
```

Comment:
What about moving this section before the "Service Tasks with attached Java code" section? 

External tasks are our preference here.

Reply:
Great idea!

Reply:
Done in [b150565](https://github.com/camunda/camunda-platform-docs/pull/2283/commits/b1505650fb63fbae4a0734444454812db625c0c8)